packages/graph/hash_graph/lib/graph/src/types/schema/array.rs

```diff
@@ -0,0 +1,313 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+enum ArrayTypeTag {
```

Comment:
When you know the pattern, these make sense. I think we should either write docs in schema/mod.rs, or perhaps just on each of these TypeTags quickly explain their purpose. (Something along the lines of saying this is to behave as a constant value)

Reply:
Added a documentation to tye `_TypeTag` types: 99c12d9c35303808bb2bf24c9cb944ea20a3f9e0

Reply:
It should be serialize as a constant _value_ right? They are not the key (to be fair the "field" could kind of be either but that's why I'd like to disambiguate)

Reply:
Good point: a1d30bb1f0e9083c110fdfa222c265ae47d06e87